### PR TITLE
KAFKA-9567: Docs, system tests for ZooKeeper 3.5.7

### PR DESCRIPTION
**Pull request by rondagostino**
These changes depend on [KIP-515: Enable ZK client to use the new TLS supported authentication](https://cwiki.apache.org/confluence/display/KAFKA/KIP-515%3A+Enable+ZK+client+to+use+the+new+TLS+supported+authentication), which was only added to 2.5.0. The upgrade to ZooKeeper 3.5.7 was merged to both 2.5.0 and 2.4.1 via https://issues.apache.org/jira/browse/KAFKA-9515, but this change must only be merged to 2.5.0 (it will break the system tests if merged to 2.4.1).

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by rondagostino** [comment]
@ijuma Take a look.  Just cannot be merged to 2.4.1 -- can only merge to 2.5.0.

**Comment by ijuma** [comment]
ok to test

**Comment by ijuma** [comment]
Since @rajinisivaram reviewed the original change, I think she may be better places to review this one too.

**Comment by rondagostino** [comment]
@omkreddy @rajinisivaram Any possibility of reviewing this?  A 2.5 RC is going to be cut soon, and this needs to be in the final release.  It is just doc changes and an additional system test.

**Comment by omkreddy** [comment]
@rondagostino Will review today. thanks.

File: tests/kafkatest/tests/core/zookeeper_tls_encrypt_only_test.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+from kafkatest.utils import is_int
+
+import logging
+
+class ZookeeperTlsEncryptOnlyTest(ProduceConsumeValidateTest):
+    """Tests TLS encryption-only (ssl.clientAuth=none) connectivity to zookeeper.
+    """
+
+    def __init__(self, test_context):
+        super(ZookeeperTlsEncryptOnlyTest   , self).__init__(test_context=test_context)
```
Comment (omkreddy) [inline tests/kafkatest/tests/core/zookeeper_tls_encrypt_only_test.py]: nit: extra space

File: tests/kafkatest/services/kafka/kafka.py [hunk 2]
```diff
@@ -435,8 +435,8 @@ def create_topic(self, topic_cfg, node=None):
         kafka_topic_script = self.path.script("kafka-topics.sh", node)
 
         cmd = kafka_topic_script + " "
-        cmd += "--zookeeper %(zk_connect)s --create --topic %(topic)s " % {
-                'zk_connect': self.zk_connect_setting(),
+        cmd += "--bootstrap-server %(bootstrap_server)s --create --topic %(topic)s " % {
```
Comment (omkreddy) [inline tests/kafkatest/services/kafka/kafka.py]: Currently we are not passing required security configs (using --command-config) to the tool.  This change may not work for with secure broker listeners.  seems like we are not using these methods in any security enabled tests.

**Review by omkreddy — APPROVED** [review]
@rondagostino Thanks for the PR. LGTM.

Can we run the full system test run with this change?

File: tests/kafkatest/services/kafka/kafka.py (hunk 2 above)
Comment (rondagostino) [reply tests/kafkatest/services/kafka/kafka.py]: Good catch!  I fixed it by having each Kafka broker create a (potentially empty, for `PLAINTEXT`) `command_config.properties` file whenever it starts; every command that uses `--bootstrap-server` also now sets `--command-config` and exports any necessary SASL-related system properties.  I successfully ran all three tests `zookeeper_tls_encrypt_only_test`, `zookeeper_tls_test`, and `zookeeper_security_upgrade_test`.  I think the full system test suite should pass with these changes.

File: tests/kafkatest/services/kafka/kafka.py [hunk 3]
```diff
@@ -94,7 +96,7 @@ class KafkaService(KafkaPathResolverMixin, JmxMixin, Service):
 
     def __init__(self, context, num_nodes, zk, security_protocol=SecurityConfig.PLAINTEXT, interbroker_security_protocol=SecurityConfig.PLAINTEXT,
                  client_sasl_mechanism=SecurityConfig.SASL_MECHANISM_GSSAPI, interbroker_sasl_mechanism=SecurityConfig.SASL_MECHANISM_GSSAPI,
-                 authorizer_class_name=None, topics=None, version=DEV_BRANCH, jmx_object_names=None,
+                 authorizer_class_name=None, topics=None, project="kafka", dist_version=None, version=DEV_BRANCH, jmx_object_names=None,
```
Comment (omkreddy) [inline tests/kafkatest/services/kafka/kafka.py]: is this change required?

**Review by omkreddy — COMMENTED** [review]
@rondagostino Thanks for the updates. LGTM. 

will start a system test run.

**Review by andrewchoi5 — APPROVED** [review]

**Comment by omkreddy** [comment]
Merging to trunk and 2.5.
